### PR TITLE
Add workflow for testing contrib repo

### LINKED ISSUE #199: CI should include running collector-contrib tests (and eventually collector tests)
Several components of the contrib collector depend on this library. Any changes made to this library have the potential to break functionality in those components. 

CI for this repo should include a workflow that runs tests with the proposed changes, for any components that depend on this library. See [this workflow](https://github.com/open-telemetry/opentelemetry-collector/blob/main/.github/workflows/contrib-tests.yml) as an example.

### PR DESCRIPTION
**Pull request by prasadkatti**
Fixes #199

### CODE REVIEW

**Comment by prasadkatti** [comment]
@djaglowski - can you please review this PR?

**Review by djaglowski — COMMENTED** [review]
Thanks for the contribution @prasadkatti!

This looks great, except it looks like we need to run tests a little differently. Currently, it is only running tests for packages in the root module.

```
go test -race -timeout 30s ./...
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontribcol	0.182s
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/internal/version	0.023s [no tests to run]
make[1]: Leaving directory '/tmp/opentelemetry-collector-contrib'
Restoring contrib to no longer use this core checkout
```

Technically, we don't need to run tests in all modules, since most do not use this library as a dependency. If doing so proves to be too time consuming, or if you want to go straight for an optimized solution, it would be nice to run tests only in the modules which list this library as a dependency.

**Comment by djaglowski** [comment]
@prasadkatti Would you be willing to update this PR according to Tigran's suggestion on the contrib PR?

> if you really need to run just a subset then manually execute the commands that are in the make target in this PR

**Comment by prasadkatti** [comment]
Yes, I will try it out and reach out if I need help.

**Comment by djaglowski** [comment]
@prasadkatti Thank you for the updates! The script looks great, and appears to be finding the correct modules, but unfortunately it's crashing on `go mod tidy` due to a recent version change. The check reported success, but logged the following:

```
Run contrib_path=/tmp/opentelemetry-collector-contrib
  contrib_path=/tmp/opentelemetry-collector-contrib
  git clone https://github.com/open-telemetry/opentelemetry-collector-contrib.git $contrib_path
  ./check-contrib.sh
  shell: sh -e {0}
Cloning into '/tmp/opentelemetry-collector-contrib'...
Log collection modules - /tmp/opentelemetry-collector-contrib
/tmp/opentelemetry-collector-contrib/cmd/configschema
/tmp/opentelemetry-collector-contrib/internal/stanza
/tmp/opentelemetry-collector-contrib/receiver/filelogreceiver
/tmp/opentelemetry-collector-contrib/receiver/syslogreceiver
/tmp/opentelemetry-collector-contrib/receiver/tcplogreceiver
/tmp/opentelemetry-collector-contrib/receiver/udplogreceiver
/tmp/opentelemetry-collector-contrib /__w/opentelemetry-log-collection/opentelemetry-log-collection
go mod tidy: go.mod file indicates go 1.17, but maximum supported version is 1.16
go: cloud.google.com/go/monitoring@v0.1.0: missing go.sum entry; to add it:
	go mod download cloud.google.com/go/monitoring
/__w/opentelemetry-log-collection/opentelemetry-log-collection
/tmp/opentelemetry-collector-contrib/cmd/configschema /__w/opentelemetry-log-collection/opentelemetry-log-collection
go mod tidy: go.mod file indicates go 1.17, but maximum supported version is 1.16
...
```

**Comment by prasadkatti** [comment]
@djaglowski Yes, I noticed that. Can you give me some pointers on how I could fix that. The container image I used in the github actions workflow points to `image: cimg/go:1.16`. So what exactly is the issue here?

File: .github/workflows/contrib-tests.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  contrib_tests:
+    runs-on: ubuntu-latest
+    container:
+      image: cimg/go:1.16
+    steps:
+      - name: Setup Permissions
+        run: sudo chmod -R 777 $GITHUB_WORKSPACE /github /__w/_temp
+      - name: Checkout Repo
+        uses: actions/checkout@v2
```
Comment (djaglowski) [inline .github/workflows/contrib-tests.yml:18]:
```suggestion
runs-on: ubuntu-latest
    steps:
      - name: Checkout Repo
        uses: actions/checkout@v2
      - name: Setup Go
        uses: actions/setup-go@v2
        with:
          go-version: 1.17
```
I think we need to run w/ Go 1.17. Likely this change would work, as we use this setup in other actions too.

**Comment by djaglowski** [comment]
I'm still seeing failures relating to go version, but the check passes. I believe we need the script to return a non-zero exit code when it fails.

As for the go version issues, it looks like the upgrade to 1.17 happened after you created your branch. Can you please rebase?

**Comment by prasadkatti** [comment]
@djaglowski - rebased on upstream and pushed the changes. Can you please kick off the workflow?

**Comment by djaglowski** [comment]
@prasadkatti Thanks for iterating on this. Looks like it's almost there. I'm happy to see that the failure registers now.

I think it's ok to exclude specifically exclude the root module.

**Comment by djaglowski** [comment]
Looks good, thanks @prasadkatti! This is a really nice improvement for this repo.

```
PASS
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/internal/stanza	0.239s
~/work/opentelemetry-log-collection/opentelemetry-log-collection
/tmp/opentelemetry-collector-contrib/receiver/filelogreceiver ~/work/opentelemetry-log-collection/opentelemetry-log-collection
go: downloading github.com/observiq/nanojack v0.0.0-20201106172433-343928847ebc
go: downloading github.com/bmatcuk/doublestar/v3 v3.0.0
go: downloading go.opencensus.io v0.23.0
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/cenkalti/backoff/v4 v4.1.1
go: downloading go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.22.0
go: downloading github.com/rs/cors v1.8.0
go: downloading go.opentelemetry.io/contrib v0.22.0
go: downloading go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.22.0
go: downloading github.com/shirou/gopsutil v3.21.8+incompatible
go: downloading github.com/felixge/httpsnoop v1.0.2
go: downloading go.opentelemetry.io/otel/metric v0.22.0
go: downloading go.opentelemetry.io/otel/internal/metric v0.22.0
go: downloading github.com/StackExchange/wmi v1.2.1
go: downloading github.com/go-ole/go-ole v1.2.5
PASS
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/filelogreceiver	0.928s
~/work/opentelemetry-log-collection/opentelemetry-log-collection
/tmp/opentelemetry-collector-contrib/receiver/syslogreceiver ~/work/opentelemetry-log-collection/opentelemetry-log-collection
go: downloading github.com/jpillora/backoff v1.0.0
go: downloading github.com/observiq/go-syslog/v3 v3.0.2
go: downloading github.com/pelletier/go-toml v1.7.0
go: downloading golang.org/x/tools v0.1.2
PASS
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/syslogreceiver	0.213s
~/work/opentelemetry-log-collection/opentelemetry-log-collection
/tmp/opentelemetry-collector-contrib/receiver/tcplogreceiver ~/work/opentelemetry-log-collection/opentelemetry-log-collection
PASS
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/tcplogreceiver	0.115s
~/work/opentelemetry-log-collection/opentelemetry-log-collection
/tmp/opentelemetry-collector-contrib/receiver/udplogreceiver ~/work/opentelemetry-log-collection/opentelemetry-log-collection
PASS
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/udplogreceiver	0.109s
~/work/opentelemetry-log-collection/opentelemetry-log-collection
```

**Review by djaglowski — APPROVED** [review]

**Pull request merged**